util/omrutil/j9memclr.c

```diff
@@ -48,9 +48,20 @@ struct IHAPSA {
 static int isZ10orGreater = -1;
 #endif
 
+#if defined(J9ZOS390) || (defined(LINUX) && defined(S390))
+#include <stdlib.h>
+#endif /* defined(J9ZOS390) || (defined(LINUX) && defined(S390)) */
+
 void
 OMRZeroMemory(void *ptr, uintptr_t length)
 {
+#if defined(J9ZOS390) || (defined(LINUX) && defined(S390))
+	static BOOLEAN useJ9zerz10Assembly = FALSE;
+	if (NULL != getenv("OMR_USE_J9ZERZ10")) {
+		useJ9zerz10Assembly = TRUE;
+	}
```

Comment:
This will get called multiple times. You need to initialize the static variable at definition to ensure `getenv` only gets called once. Here is how to do it:

```
static BOOLEAN useJ9zerz10Assembly = (NULL != getenv("OMR_USE_J9ZERZ10"));
```

Reply:
Uuups just saw your earlier message about the compiler not accepting the above as a constant. Not sure how to work around that, but we definitely don't want to be calling `getenv` every time. That will kill performance.

Reply:
Switching to C++ is probably the best idea. Alternatively, we could just implement this code as:

```c++
	static int useJ9zerz10Assembly = -1;
	if (-1 == useJ9zerz10Assembly) {
		useJ9zerz10Assembly = (NULL != getenv("OMR_USE_J9ZERZ10"));
	}
```

Multiple threads might reach the getenv call, but they should all have a consistent view of the environment.

Do we need to worry about concurrent setenv calls here?

Reply:
> Do we need to worry about concurrent setenv calls here?

No I don't think that is a concern. I too would prefer just migrating to C++ so we can avoid these workarounds but I'm ok with implementing your suggestion as well. I'll defer to GC experts to suggest which way we go. Thanks!

Reply:
OK, we both agree that migrating this file to C++ is the best option. @amicic and @dmitripivkine, what do you guys think?

Reply:
I thinks that's probably why `isZ10orGreater` is an int instead of a boolean[1]. So moving to C++ would allow us to change that to a boolean as well.

1 : https://github.com/eclipse/omr/blob/aa598e54a99032ef10213afeb7798c4140066c15/util/omrutil/j9memclr.c#L134

Reply:
@amicic and @dmitripivkine should we migrate [util/omrutil/j9memclr.c](https://github.com/eclipse/omr/blob/master/util/omrutil/j9memclr.c) to C++?

Reply:
I was just on a call with @amicic and I can confirm he gave us blessing to migrate this file to C++. @VermaSh could you go ahead and draft that change?